packages/app/obojobo-express/server/routes/api/visits.js

```diff
@@ -91,6 +99,26 @@ router
 					visitId: req.currentVisit.id
 				})
 			})
+			.then(() => viewerUsersState.getLastLogin(userId))
+			.then(result => {
+				if (result) lastLogin = result.last_login
+			})
+			.then(() => viewerNotificationState.getId(lastLogin))
+			.then(result => {
+				if (result) id = result.id
+			})
+			.then(() => viewerNotificationState.getStatus(id))
+			.then(result => {
+				if (result) isNotificationEnabled = result.status
+			})
+			.then(() => viewerNotificationState.getTitle(id))
+			.then(result => {
+				if (result) notificationTitle = result.title
+			})
+			.then(() => viewerNotificationState.getText(id))
+			.then(result => {
+				if (result) notificationText = result.text
+			})
```

Comment:
Try to provide all valid notifications to the frontend in a single array rather than going through multiple steps to grab individual pieces of a single notification like this.